common/logisticspipes/gui/guidebook/book/MenuItem.java

```diff
@@ -0,0 +1,127 @@
+import net.minecraft.client.audio.PositionedSoundRecord;
+import net.minecraft.client.audio.SoundHandler;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.client.renderer.RenderHelper;
+import net.minecraft.init.SoundEvents;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ResourceLocation;
+
+import lombok.Getter;
+
+import logisticspipes.LPConstants;
+import logisticspipes.gui.guidebook.GuiGuideBook;
+import logisticspipes.utils.GuideBookContents;
+import logisticspipes.utils.string.StringUtils;
+
+public class MenuItem {
+
+	private static final ResourceLocation GUI_BOOK_TEXTURE = new ResourceLocation(LPConstants.LP_MOD_ID, "textures/gui/guide_book.png");
+
+	// Getting constants
+	private final int z$titleButtons = 15; // Title and Buttons Z
+	private final int z$frame = 10; // Frame Z
+	private final int z$text = 5;// Text/Information Z
+	private final int z$background = 0;  // Background Z
+
+	//// Information storage
+	@Getter
+	private GuideBookContents.Chapter chapter;
+
+	//// Drawing variables
+	public boolean visiible, hovering, enabled;
+	private int btn$bgX0, btn$bgY0, btn$bgX1, btn$bgY1;
+	private int btn$x0, btn$y0, btn$x1, btn$y1, btn$x2, btn$y2, btn$x3, btn$y3;
+
+	//// Atlas
+	// Btn
+	private final int atlas$btn$bgU0 = 64, atlas$btn$bgV0 = 32, atlas$btn$bgU1 = 96, atlas$btn$bgV1 = 64;
+	private final int atlas$btn$u0 = 0, atlas$btn$v0 = 64, atlas$btn$u1 = 2, atlas$btn$v1 = 66, atlas$btn$u2 = 14, atlas$btn$v2 = 78, atlas$btn$u3 = 16, atlas$btn$v3 = 80;
+
+	public MenuItem(GuideBookContents.Chapter chapter) {
+		this.chapter = chapter;
+		//// Drawing stuff
+		this.visiible = true;
+		this.hovering = false;
+		this.enabled = true;
+	}
+
+	public void drawMenuItem(Minecraft mc, int mouseX, int mouseY, int x, int y, int sizeX, int sizeY, boolean text) {
+		GlStateManager.color(1.0F, 1.0F, 1.0F);
+		drawMenuItemFrame(mc, mouseX, mouseY, x, y, sizeX, sizeY);
+		int icon$sizeX, icon$sizeY, icon$offSetX, icon$offSetY;
+		double icon$scaleX, icon$scaleY;
+		icon$scaleX = 1.0;
+		icon$scaleY = 1.0;
+		icon$sizeX = 16 * (int) icon$scaleX;
+		icon$sizeY = 16 * (int) icon$scaleY;
+		icon$offSetX = (sizeX - icon$sizeX) / 2;
+		icon$offSetY = (sizeY - icon$sizeY) / 2;
+		GlStateManager.pushMatrix();
+		GlStateManager.translate(x + icon$offSetX, y + icon$offSetY, z$text);
+		GlStateManager.scale(icon$scaleX, icon$scaleY, 0);
+		RenderHelper.enableGUIStandardItemLighting();
+		mc.getRenderItem().renderItemAndEffectIntoGUI(new ItemStack(Item.REGISTRY.getObject(new ResourceLocation(chapter.getItem()))), 0, 0);
+		RenderHelper.disableStandardItemLighting();
+		GlStateManager.scale(1 / icon$scaleX, 1 / icon$scaleY, 0);
+		GlStateManager.popMatrix();
+	}
+
+	public void drawMenuItemFrame(Minecraft mc, int mouseX, int mouseY, int x, int y, int sizeX, int sizeY) {
+		mc.renderEngine.bindTexture(GUI_BOOK_TEXTURE);
+		{
+			btn$bgX0 = x + 1;
```

Comment:
why do you need a class property for this? it is only set to `x + 1` here and then used in the function call below. Please check this with all class variables for the next review round.

Reply:
Its just to be easier for me to spot rendering problems, and I've had plenty of those, once all the difficult part is dealt with I'll go around removing useless variables